### PR TITLE
Converting from browserPath to browserFlavor

### PR DESCRIPTION
**Pull request by vidorteg**
This PR removes the _browserPath setting_ and creates a _browser flavor combo box_ for a similar purpose.

image

**Current Behavior:**
The user can specify (via a input box) the path to a browser version, when no path is specified the extension searches for stable > beta > dev > canary (in that order) until an existing version is found on disk.

**New Behavior**
When default is selected, extension will keep current behavior, otherwise it will use the version specified by the user. 

Test are also modified to use a specific flavor instead of a custom path.

### CODE REVIEW

**Comment by mliao95** [comment]
Just did a quick first pass and it works well!  One NIT would be to just capitalize the options.  Will take a closer look tomorrow.

**Comment by vidorteg** [comment]
> Just did a quick first pass and it works well! One NIT would be to just capitalize the options. Will take a closer look tomorrow.

done :)

File: src/utils.ts [hunk 1]
```diff
@@ -261,32 +256,25 @@ export function getPlatform(): Platform {
...
-        for (const p of searchPaths) {
-            if (await fse.pathExists(p)) {
-                return p;
-            }
+    const flavor: BrowserFlavor | undefined = config.browserFlavor || settings.get("browserFlavor");
+
+    switch (getPlatform()) {
+        case "Windows": {
+           return await verifyFlavorPath(flavor, "Windows");
+           break;
```
Comment (mliao95) [inline src/utils.ts:269]: We don't need break statements if we are returning the line before.

File: src/utils.ts [hunk 2]
```diff
@@ -225,7 +220,7 @@ export function getRemoteEndpointSettings(config: Partial<IUserConfig> = {}): ID
         }
     }
 
-    if (userDataDir === true || (typeof userDataDir === "undefined" && !browserPath)) {
+    if (userDataDir === true || (typeof userDataDir === "undefined" && browserPathSet)) {
```
Comment (mliao95) [inline src/utils.ts:223]: Correct me if I am wrong, but I think we either want to flip the boolean here or in browserPathSet above.

browserPath would return false if the user used default and would return true if the user had an alternate path
browserPathSet would also return false if the browserFlavor was default and return true if the user set it to anything else but default

It looks like these booleans return true/false for default/not default, so we shouldn't remove the "!" from the if statement.

Comment (vidorteg) [reply src/utils.ts]: good catch!

File: src/utils.ts (hunk 1 above)
Comment (vidorteg) [reply src/utils.ts]: 😱, how did the linter not catch that one?! nice one Michael.

**Pull request merged**